gulpfile.coffee

```diff
@@ -1,69 +1,116 @@
+)
 main_bower_files = require 'main-bower-files'
 del = require 'del'
 exec = require('child_process').exec
-minimist = require 'minimist'
-
 
 root_dir = './main'
 static_dir = "#{root_dir}/static"
+bower_dir = "#{static_dir}/bower_components"
+style_dir = "#{static_dir}/src/style"
+script_dir = "#{static_dir}/src/script"
+
+# We handpick the paths in the order we want then to be injected or
+# concatenated, when minified.
 paths =
+  styles: [
+    "#{static_dir}/dst/style/style.css"
+  ]
+  scripts: [
+    "#{static_dir}/dst/script/common/service.js"
+    "#{static_dir}/dst/script/common/util.js"
+    "#{static_dir}/dst/script/site/app.js"
+    "#{static_dir}/dst/script/site/admin.js"
+    "#{static_dir}/dst/script/site/profile.js"
+    "#{static_dir}/dst/script/site/auth.js"
+    "#{static_dir}/dst/script/site/user.js"
+  ]
+  bower_components: [
+    "#{bower_dir}/jquery/dist/jquery.js"
+    "#{bower_dir}/moment/moment.js"
+    "#{bower_dir}/nprogress/nprogress.js"
+    "#{bower_dir}/bootstrap-sweetalert/lib/sweet-alert.js"
+    "#{bower_dir}/bootstrap/js/alert.js"
+    "#{bower_dir}/bootstrap/js/button.js"
+    "#{bower_dir}/bootstrap/js/transition.js"
+    "#{bower_dir}/bootstrap/js/collapse.js"
+    "#{bower_dir}/bootstrap/js/dropdown.js"
+    "#{bower_dir}/bootstrap/js/tooltip.js"
+  ]
   clean: [
-      "#{static_dir}/dst"
-      "#{static_dir}/ext"
-      "#{static_dir}/min"
-    ]
+    "#{static_dir}/dst"
+    "#{static_dir}/min"
+  ]
   watch: [
-      "#{static_dir}/**/*.css"
-      "#{static_dir}/**/*.js"
-      "#{root_dir}/**/*.html"
-      "#{root_dir}/**/*.py"
-    ]
-
-
-run = (option) ->
-  proc = exec "python -u run.py -#{option}"
-  proc.stderr.on 'data', (data) -> process.stderr.write data
-  proc.stdout.on 'data', (data) -> process.stdout.write data
+    "#{static_dir}/dst/style/**/*.css"
+    "#{static_dir}/dst/script/**/*.js"
+    "#{root_dir}/**/*.html"
+    "#{root_dir}/**/*.py"
+  ]
 
 
 gulp.task 'clean', ->
   del paths.clean
 
+
 gulp.task 'bower_install', ->
   $.bower()
 
-gulp.task 'ext', ['bower_install'], ->
-  gulp.src(main_bower_files(),
-    base: 'bower_components'
-  ).pipe gulp.dest("#{static_dir}/ext")
+
+# Building the style, styles from bower_components are imported in style.less
+gulp.task 'less', ->
+  gulp.src("#{style_dir}/style.less")
+    .pipe $.plumber()
+    .pipe $.less()
+    .pipe gulp.dest("#{static_dir}/dst/style/")
+    .pipe $.min_css(keepBreaks: true)
+    .pipe gulp.dest("#{static_dir}/min/style/")
+
+
+gulp.task 'coffee', ->
+  gulp.src("#{script_dir}/**/*.coffee")
+    .pipe $.plumber()
+    .pipe $.coffee(bare:true)
+    .pipe gulp.dest("#{static_dir}/dst/script/")
+
+  gulp.src(paths.scripts)
+    .pipe $.concat("script.min.js")
+    .pipe $.uglify()
+    .pipe gulp.dest("#{static_dir}/min/script/")
+
+  gulp.src(paths.bower_components)
+    .pipe $.plumber()
+    .pipe $.concat("libs.min.js")
+    .pipe $.uglify()
+    .pipe gulp.dest("#{static_dir}/min/script/")
```

Comment:
And this part is where I don't like.. :( but `script`, `libs` are very much hardcoded here and it will be hard to split things in different groups

I agree that most likely in practice you don't need more than two or it could be all one.. but nevertheless I'm guessing some projects require more..